src/Workspaces/Core/Portable/Workspace/Solution/RecoverableTextAndVersion.cs

```diff
@@ -57,11 +57,9 @@ private bool TryGetInitialSourceOrRecoverableText([NotNullWhen(true)] out ITextA
         public bool TryGetValue(LoadTextOptions options, [MaybeNullWhen(false)] out TextAndVersion value)
         {
             if (TryGetInitialSourceOrRecoverableText(out var source, out var recoverableText))
-            {
                 return source.TryGetValue(options, out value);
-            }
 
-            if (recoverableText.TryGetValue(out var text) && recoverableText.LoadTextOptions == options)
+            if (recoverableText.LoadTextOptions == options && recoverableText.TryGetValue(out var text))
```

Comment:
reversed order.  matches code in the method below, and allows us to fast path out.